File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    int [] visited = new int[201];

    public void dfs(int a, int n, int[][] computers){
        for(int i=0; i<n; i++){
            if(a!=i && computers[a][i]==1){
                if(visited[i]==0){
                    visited[i]=1;
                    dfs(i,n,computers);
                }
            }
        }
    }

    public int solution(int n, int[][] computers) {
        int answer = 0;
        for(int i=0; i<n; i++) visited[i] = 0;
        for(int i=0; i<n; i++) {
            if(visited[i]==0){
                answer++; dfs(i, n,computers);
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <queue>
#include <vector>

using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    vector<bool> visited(n); //컴퓨터가 0~n-1
    
    queue<int> q;
    for(int i=0; i<n; i++){
        if(!visited[i]){
            q.push(i);
            while(!q.empty()){
                int cur = q.front();
                q.pop();
                
                for(int j=0; j<n; j++){
                    if(computers[cur][j]==1 && !visited[j]){
                        visited[j]=true;
                        q.push(j);
                    }
                }
            }
            answer++;
        }
    }
    
    return answer;
}